pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/KitRequestShipping.java

```diff
@@ -1357,8 +1357,44 @@ public static Address getToAddressId(@NonNull EasyPostUtil easyPostUtil, KitRequ
             if (StringUtils.isNotBlank(addressId)) {
                 toAddress = easyPostUtil.getAddress(addressId);
             } else if (participant != null) {
+                boolean hasCareOf = kitRequestSettings.getHasCareOF() == 1;
+                if (!hasCareOf) { // aside form singular, all other studies should go here and proceed with normal label
+                    toAddress = easyPostUtil.createAddress(participant, kitRequestSettings.getPhone());
+                    return toAddress;
+                }
+                toAddress = getAddressForStudiesWithCareOfField(easyPostUtil, kitRequestSettings, participant, ddpInstanceDto);
+            }
+        }
+        return toAddress;
+    }
+
+    private static Address getAddressForStudiesWithCareOfField(@NonNull EasyPostUtil easyPostUtil, KitRequestSettings kitRequestSettings,
+                                                               DDPParticipant participant, DDPInstanceDto ddpInstanceDto)
+            throws EasyPostException {
+        Address toAddress = null;
+        String proxyFirstName = null;
+        String proxyLastName = null;
+        Map<String, Map<String, Object>> participantESData = ElasticSearchUtil.getFilteredDDPParticipantsFromES(ddpInstanceDto,
+                ElasticSearchUtil.BY_GUID + participant.getParticipantId());
+        if (participantESData != null && !participantESData.isEmpty()) {
+            ArrayList<String> proxies =
+                    (ArrayList<String>) participantESData.get(participant.getParticipantId()).get(ElasticSearchUtil.PROXIES);
+            if (proxies != null && proxies.size() > 0) {
+                String proxyParticipantId = proxies.get(0);
+                Map<String, Map<String, Object>> proxyESData = ElasticSearchUtil.getFilteredDDPParticipantsFromES(ddpInstanceDto,
+                        ElasticSearchUtil.BY_GUID + proxyParticipantId);
+                Map<String, Object> proxyProfile = (Map<String, Object>) proxyESData.get(proxyParticipantId)
+                        .get(ElasticSearchUtil.PROFILE);
+                proxyFirstName = (String) proxyProfile.get("firstName");
+                proxyLastName = (String) proxyProfile.get("lastName");
+                String careOf = String.format("C/O %s %s", Objects.toString(proxyFirstName, ""),
+                        Objects.toString(proxyLastName, ""));
+                toAddress = easyPostUtil.createAddress(participant, kitRequestSettings.getPhone(), careOf);
+            } else { // participant doesn't have proxies, proceed with normal label
                 toAddress = easyPostUtil.createAddress(participant, kitRequestSettings.getPhone());
             }
+        } else { // participant is not in ES
+            toAddress = easyPostUtil.createAddress(participant, kitRequestSettings.getPhone());
```

Comment:
@SimoneMaiwald not sure what should happen here, we don't have pt in ES but we have DDPParticipant

Reply:
the nothing. label creation with info available

Reply:
ok good so this is correct

Reply:
wait, that is for the singular way? I guess all pts what get a label in singular should be in ES. So I would say error log and no label